File: api/app/infrastructure/in_memory_route_repository.rb
# frozen_string_literal: true

class InMemoryRouteRepository
  def initialize(routes = [])
    @routes = routes
  end

  def find_all
    @routes
  end

  def save(route)
    @routes << route
  end

  def reset_state
    @routes = []
  end
end

File: web-app/src/infrastructure/InMemoryRouteRepository.ts
import {Route} from "@/domain/Route";
import {RouteRepository} from "@/domain/RouteRepository";

export class InMemoryRouteRepository implements RouteRepository {
  private readonly routes: Route[];

  constructor(routes: any[] = []) {
    this.routes = routes;
  }

  async save(route: Route): Promise<void> {
    this.routes.push(route);
  }

  async search(): Promise<Route[]> {
    return this.routes;
  }
}
